### PR TITLE
NT-33: Display number of left cards

### PR DESCRIPTION
**Pull request by Steellemm**
📝 **NT-33**

# Description

Display number of left cards

### CODE REVIEW

File: src/main/kotlin/org/expo/nothanks/model/event/output/RoundStartedMessage.kt [hunk 1]
```diff
@@ -3,7 +3,8 @@ package org.expo.nothanks.model.event.output
 data class RoundStartedMessage(
     val currentPlayerNumber: Int,
     val players: List<SafeGamePLayer>,
-    val currentCard: Int
+    val currentCard: Int,
+    val leftNumberCards: Int
```
Comment (TorgashSad) [inline src/main/kotlin/org/expo/nothanks/model/event/output/RoundStartedMessage.kt:7]: давай remaining, надо
либо remainingCards или numberOfRemainingCards

File: src/main/kotlin/org/expo/nothanks/model/event/output/TakeCardMessage.kt [hunk 2]
```diff
@@ -4,5 +4,6 @@ data class TakeCardMessage(
     val playerNumber: Int,
     val takenCard: Int,
     val newCardNumber: Int,
-    val allPlayerCards: Set<Int>
+    val allPlayerCards: Set<Int>,
+    val leftNumberCards: Int
```
Comment (TorgashSad) [inline src/main/kotlin/org/expo/nothanks/model/event/output/TakeCardMessage.kt:8]: тоже

File: src/main/kotlin/org/expo/nothanks/service/NotificationService.kt [hunk 3]
```diff
@@ -56,7 +56,8 @@ class NotificationService(
         messageToTopic(lobby, RoundStartedMessage(
             currentPlayerNumber = lobby.getGame().currentPlayerNumber(),
             players = lobby.getPlayersInGame(),
-            currentCard = lobby.getGame().currentCard()
+            currentCard = lobby.getGame().currentCard(),
+            leftNumberCards = lobby.getGame().leftNumberCards()
```
Comment (TorgashSad) [inline src/main/kotlin/org/expo/nothanks/service/NotificationService.kt:60]: тоже

File: src/main/kotlin/org/expo/nothanks/service/NotificationService.kt [hunk 4]
```diff
@@ -73,7 +74,8 @@ class NotificationService(
             playerNumber = lobby.getGame().currentPlayerNumber(),
             takenCard = lobby.getGame().previousCard(),
             newCardNumber = lobby.getGame().currentCardNumber(),
-            allPlayerCards = lobby.getGame().currentPlayer().cards
+            allPlayerCards = lobby.getGame().currentPlayer().cards,
+            leftNumberCards = lobby.getGame().leftNumberCards()
```
Comment (TorgashSad) [inline src/main/kotlin/org/expo/nothanks/service/NotificationService.kt:78]: тоже

File: src/main/kotlin/org/expo/nothanks/utils/GameUtils.kt [hunk 5]
```diff
@@ -87,6 +87,10 @@ fun calculateCards(cards: Set<Int>): Int {
     return score
 }
 
+fun Game.leftNumberCards(): Int {
```
Comment (TorgashSad) [inline src/main/kotlin/org/expo/nothanks/utils/GameUtils.kt:90]: тоже

File: src/main/resources/static/js/main.js [hunk 6]
```diff
@@ -89,11 +89,13 @@ function processTopicMessage(message) {
     }
     if (message['type'] === "RoundStartedMessage") {
         renderGameScreen(message.players, message['currentCard'], message['currentPlayerNumber'])
+        updateLeftNumberCards(message['leftNumberCards'])
```
Comment (TorgashSad) [inline src/main/resources/static/js/main.js:92]: тоже

File: src/main/resources/static/js/main.js [hunk 7]
```diff
@@ -89,11 +89,13 @@ function processTopicMessage(message) {
...
     }
     if (message['type'] === "TakeCardMessage") {
         if (message['playerNumber'] !== myNumber) {
             updateCardsForPlayer(message['playerNumber'],  message['allPlayerCards'])
         }
+        updateLeftNumberCards(message['leftNumberCards'])
```
Comment (TorgashSad) [inline src/main/resources/static/js/main.js:98]: тоже

File: src/main/resources/static/js/render.js [hunk 8]
```diff
@@ -134,6 +134,10 @@ function updateCardsForPlayer(playerNumber, cards) {
     $(`#other-player-card-block-${playerNumber}`).html(renderCards(cards))
 }
 
+function updateLeftNumberCards(number) {
```
Comment (TorgashSad) [inline src/main/resources/static/js/render.js:137]: тоже

**Review by TorgashSad — CHANGES_REQUESTED** [review]
Left to remaining

File: src/main/kotlin/org/expo/nothanks/model/event/output/RoundStartedMessage.kt (hunk 1 above)
Comment (iamsaywhat) [reply src/main/kotlin/org/expo/nothanks/model/event/output/RoundStartedMessage.kt:7]: I vote for `remainingCards`

**Review by iamsaywhat — APPROVED** [review]
I have nothing to add to @TorgashSad's comments, so I approve

**Review by TorgashSad — APPROVED** [review]
эх ты

**Pull request merged**